### PR TITLE
mds: remove the incorrect std::move for fsname and path 

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <xiubli@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by leonid-s-usov — APPROVED** [review]
TBH, I don't understand why `move` is even allowed with a `const &` argument, but my tests show that a copy constructor is called despite the move.

If we would like to allow for a move semantics and avoid creating two versions of the constructor (or 4) for the permutation of `const &` and `&&` arguments, here's a c++20 way to declare it:

```
  template<std::convertible_to<std::string> N, std::convertible_to<std::string> P>
  MDSCapMatch(N&& fsname_, P&& path_,
	      bool root_squash_, int64_t uid_=MDS_AUTH_UID_ANY,
	      const std::vector<gid_t>& gids_={})
{
    fs_name = std::forward<N>(fsname_);
    path = std::forward<P>(path_);
...
```

declared this way it can be called as 

```
MDSCapMatch(name, path, ...);

MDSCapMatch(std::move(name), path, ...);

MDSCapMatch("literal name", path, ...);
```

etc

**Comment by batrick** [comment]
> TBH, I don't understand why `move` is even allowed with a `const &` argument, but my tests show that a copy constructor is called despite the move.

My understanding is that it will create a `std::string const&&` in this instance. (You could have a constructor/operator that accepts such a type.)

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
I'm also not sure how this PR fixes https://tracker.ceph.com/issues/65563 which seems to be a kclient issue?

File: src/mds/MDSAuthCaps.h [hunk 1]
```diff
@@ -118,8 +118,8 @@ struct MDSCapMatch {
   MDSCapMatch(const std::string& fsname_, const std::string& path_,
 	      bool root_squash_, int64_t uid_=MDS_AUTH_UID_ANY,
 	      const std::vector<gid_t>& gids_={}) {
-    fs_name = std::move(fsname_);
-    path = std::move(path_);
+    fs_name = fsname_;
```
[the remark this answers is not in the document]
Comment (lxbsz) [reply src/mds/MDSAuthCaps.h:121]: yeah, I think I found the root cause. It's the bug of the `git_t`, actually it's 32 bits while the kclient is trying to decode it as `int64_t`.

**Comment by lxbsz** [comment]
> I'm also not sure how this PR fixes https://tracker.ceph.com/issues/65563 which seems to be a kclient issue?

Actaully this is not a correct fix, please see https://github.com/ceph/ceph/pull/56981#discussion_r1570820276.

**Review by batrick — APPROVED** [review]

**Review by batrick — CHANGES_REQUESTED** [review]
please remove the `Fixes` line in `mds: remove the imcorrect std::move for fsname and path`

**Comment by lxbsz** [comment]
> please remove the `Fixes` line in `mds: remove the imcorrect std::move for fsname and path`

Done. Thanks!

**Comment by leonid-s-usov** [comment]
@lxbsz with the next push, would you please fix the typo in the commit summary 07b01abf6691074ea876b3b5c83f2e00203c0dea? `imcorrect` -> `incorrect`

**Comment by lxbsz** [comment]
> @lxbsz with the next push, would you please fix the typo in the commit summary [07b01ab](https://github.com/ceph/ceph/commit/07b01abf6691074ea876b3b5c83f2e00203c0dea)? `imcorrect` -> `incorrect`

Done. Thanks @leonid-s-usov

**Comment by batrick** [comment]
@lxbsz I think you added the ostream changes back in by accident.

**Comment by lxbsz** [comment]
> @lxbsz I think you added the ostream changes back in by accident.

Removed it.

**Comment by Svelar** [comment]
jenkins test make check arm64

**Review by leonid-s-usov — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
make check failure is unrelated

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
jenkins test make check

**Review by dparmar18 — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
@batrick This is under QA already.

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65771](https://tracker.ceph.com/issues/65771).

**Comment by batrick** [comment]
> @batrick This PR is under QA already.

Feel free to merge it if your QA completes successfully.

**Review by rishabh-d-dave — APPROVED** [review]
QA run was successful - https://tracker.ceph.com/issues/65764.

Testing took more time than expected because there were 25-30 new failures, most of them caused by a PR in the testing branch.

**Pull request merged**